code/modules/xenoarchaeology/traits/xenoartifact_minors.dm

```diff
@@ -0,0 +1,430 @@
+	if(istype(item, /obj/item/multitool))
+		to_chat(user, "<span class='info'>The [item.name] displays an overcharge reading of [charges/3].</span>") 
+		return TRUE
+	return ..()
+
+//============
+// Dense, makes the artifact mimic a structure
+//============
+/datum/xenoartifact_trait/minor/dense //Rather large, quite gigantic, particularly big
+	desc = "Dense"
+	label_desc = "Dense: The Artifact is dense and cannot be easily lifted but, the design has a slightly higher reaction rate."
+	blacklist_traits = list(/datum/xenoartifact_trait/minor/wearable, /datum/xenoartifact_trait/minor/sharp, /datum/xenoartifact_trait/minor/light, /datum/xenoartifact_trait/minor/heavy, /datum/xenoartifact_trait/minor/blocking, /datum/xenoartifact_trait/minor/anchor, /datum/xenoartifact_trait/minor/slippery)
+	flags = BLUESPACE_TRAIT | URANIUM_TRAIT
+
+/datum/xenoartifact_trait/minor/dense/on_init(obj/item/xenoartifact/X)
+	X.density = TRUE
+	X.interaction_flags_atom = INTERACT_ATOM_ATTACK_HAND
+	X.interaction_flags_item = INTERACT_ATOM_ATTACK_HAND
+	X.charge_req += 20
+
+//============
+// Sharp, makes the artifact do extra damage and slice type
+//============
+/datum/xenoartifact_trait/minor/sharp
+	desc = "Sharp"
+	label_desc = "Sharp: The Artifact is shaped into a fine point. Perfect for popping balloons."
+	blacklist_traits = list(/datum/xenoartifact_trait/minor/dense)
+	flags = PLASMA_TRAIT | URANIUM_TRAIT
+
+/datum/xenoartifact_trait/minor/sharp/on_touch(obj/item/xenoartifact/X, mob/user)
+	to_chat(user, "<span class='notice'>The [X.name] feels sharp.</span>")
+	return TRUE
+
+/datum/xenoartifact_trait/minor/sharp/on_init(obj/item/xenoartifact/X)
+	X.sharpness = IS_SHARP_ACCURATE
+	X.force = X.charge_req*0.12
+	X.attack_verb = list("cleaved", "slashed", "stabbed", "sliced", "tore", "ripped", "diced", "cut")
+	X.attack_weight = 2
+	X.armour_penetration = 5
+
+//============
+// Cooler, reduces cooldown times
+//============
+/datum/xenoartifact_trait/minor/cooler
+	desc = "Frosted"
+	label_desc = "Frosted: The Artifact has the unique property of actively cooling itself. This also seems to reduce time between uses."
+	flags = BLUESPACE_TRAIT | PLASMA_TRAIT | URANIUM_TRAIT
+
+/datum/xenoartifact_trait/minor/cooler/on_touch(obj/item/xenoartifact/X, mob/user)
+	to_chat(user, "<span class='notice'>The [X.name] feels cold.</span>")
+	return TRUE
+
+/datum/xenoartifact_trait/minor/cooler/on_init(obj/item/xenoartifact/X)
+	X.cooldown = 4 SECONDS //Might revisit the value.
+
+/datum/xenoartifact_trait/minor/cooler/activate(obj/item/xenoartifact/X)
+	X.charge -= 10
+
+//============
+// Sentient, allows a ghost to control the artifact
+//============
+/datum/xenoartifact_trait/minor/sentient
+	label_name = "Sentient"
+	label_desc = "Sentient: The Artifact seems to be alive, influencing events around it. The Artifact wants to return to its master..."
+	///he who lives inside
+	var/mob/living/simple_animal/man
```

Comment:
I think you should use a non abstract type like the soulstone ghost or something

Reply:
switched to soulstone ghost, shade